### PR TITLE
Lint: add w0101 and w0404 checks

### PR DESCRIPTION
**Pull request by clebergnu**
Enabling an additional couple of warnings to our lint check.

### CODE REVIEW

**Comment by clebergnu** [comment]
[Enable `W` in travis/make check.](https://trello.com/c/Ohh7Q6qj/1183-enable-w-in-travis-make-check)

File: selftests/checkall [hunk 1]
```diff
@@ -161,7 +161,7 @@ results_dir_content() {
 
 [ "$SKIP_RESULTSDIR_CHECK" ] || RESULTS_DIR_CONTENT="$(ls $RESULTS_DIR 2> /dev/null)"
 if [ "$TRAVIS" == "true" ]; then
-    run_rc lint 'RESULT=$(mktemp); inspekt lint --exclude=.git --enable W0102,W0611 &>$RESULT || { cat $RESULT; rm $RESULT; exit -1; }; rm $RESULT'
+    run_rc lint 'RESULT=$(mktemp); inspekt lint --exclude=.git --enable W0102,W0611,W0612,W0622 &>$RESULT || { cat $RESULT; rm $RESULT; exit -1; }; rm $RESULT'
```
Comment (ldoktor) [inline selftests/checkall]: shouldn't this be the last commit (when these are tackled?)

Comment (ldoktor) [reply selftests/checkall]: Oh no fixes were needed?

Comment (ldoktor) [reply selftests/checkall]: OK, I see, these were added in previous PR and were only included in `make check`...

Comment (ldoktor) [reply selftests/checkall]: OK, too tired eyes. This is the only place where we enable warnings and as it's passing there were probably no issues...

File: optional_plugins/varianter_yaml_to_mux/tests/test_mux.py [hunk 2]
```diff
@@ -127,10 +127,10 @@ def test_basic_functions(self):
                   'prod']
         self.assertEqual(leaves, self.tree.get_leaves())
         # ascii contain all leaves and doesn't raise any exceptions
-        ascii = tree.tree_view(self.tree, 0, False).decode('ascii')
+        ascii_ = tree.tree_view(self.tree, 0, False).decode('ascii')
```
Comment (ldoktor) [inline optional_plugins/varianter_yaml_to_mux/tests/test_mux.py]: I know it's just selftest, but using `${variable}_` is not really nice practice. How about about `tree_view` or just `output`?

**Comment by clebergnu** [comment]
Closing in favor of #2548

**Pull request closed without merging**